### ISSUE TITLE
LINK : fatal error LNK1104: cannot open file 'boost_python3-vc141-mt-1_65_1.lib' [C:\usd_gltf\build\usd_from_gltf\usd_from_gltf\usd_from_gltf.vcxproj] LINK : fatal error LNK1104: cannot open file 'boost_python3-vc141-mt-1_65_1.lib' [C:\usd_gltf\build\usd_from_gltf\ufg_plugin\ufg_plugin.vcxproj]    USD_FROM_GLTF: ERROR: Command failed: cmake --build . --config Release --target install -- /m   USD_FROM_GLTF: See log at: C:\usd_gltf\build\usd_from_gltf\log.txt

### ISSUE THREAD

**Issue by lanxinger** [google/usd_from_gltf#27]
Unknown compiler version - please run the configure tests and report the results

Having trouble compiling usd_from_gltf. I have USD built from source on the target machine. any idea on how to solve this error?

thanks
Markus

**Comment by C0DED00D** [comment]
What version of Visual Studio are you using? So far, we've only been building with Visual Studio 2017. 

Thanks!

**Comment by lanxinger** [comment]
yes actually that is what I am using as well. Anything else I could be forgetting. USD only builds with Python 2 but this seems to somehow ask for a python3 library?

**Comment by vanlicht** [comment]
Hi, I encounter the same challenge. I also use VS201 and Python 2.7.
Here's the error part at the stage of __Installing USD_from_GLTF ___

-------- Installing USD_FROM_GLTF --------
  USD_FROM_GLTF: CWD: D:\Git\UFG\build\usd_from_gltf-master
  USD_FROM_GLTF: Run: cmake D:\Git\UFG\usd_from_gltf-master -DCMAKE_INSTALL_PREFIX=D:\Git\UFG -DCMAKE_PREFIX_PATH=D:\Git\UFG -DUSD_DIR=D:\Git\USD\inst\USD -DCMAKE_GENERATOR_PLATFORM=x64
  USD_FROM_GLTF: CWD: D:\Git\UFG\build\usd_from_gltf-master
  USD_FROM_GLTF: Run: cmake --build . --config Release --target install -- /m
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/common/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/convert/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/gltf/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/process/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/usd_from_gltf/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/ufg_plugin/CMakeFiles/generate.stamp is up-to-date.
  Building Custom Rule D:/Git/UFG/usd_from_gltf-master/gltf/CMakeLists.txt
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/gltf/CMakeFiles/generate.stamp is up-to-date.
  internal_util.cc
  load.cc
  memory_stream.cc
  message.cc
  stream.cc
  validate.cc
  cache.cc
  disk_stream.cc
  disk_util.cc
  glb_stream.cc
  gltf.cc
  image_parsing.cc
  Generating Code...
  gltf.vcxproj -> D:\Git\UFG\build\usd_from_gltf-master\gltf\Release\gltf.lib
  Building Custom Rule D:/Git/UFG/usd_from_gltf-master/common/CMakeLists.txt
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/common/CMakeFiles/generate.stamp is up-to-date.
  config.cc
  Unknown compiler version - please run the configure tests and report the results
  logging.cc
  Unknown compiler version - please run the configure tests and report the results
  platform.cc
  Generating Code...
  common.vcxproj -> D:\Git\UFG\build\usd_from_gltf-master\common\Release\common.lib
  Building Custom Rule D:/Git/UFG/usd_from_gltf-master/process/CMakeLists.txt
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/process/CMakeFiles/generate.stamp is up-to-date.
  math.cc
  Unknown compiler version - please run the configure tests and report the results
  mesh.cc
  Unknown compiler version - please run the configure tests and report the results
  process_util.cc
  Unknown compiler version - please run the configure tests and report the results
  skin.cc
  Unknown compiler version - please run the configure tests and report the results
  animation.cc
  Unknown compiler version - please run the configure tests and report the results
  color.cc
  Unknown compiler version - please run the configure tests and report the results
  float_image.cc
  Unknown compiler version - please run the configure tests and report the results
  image.cc
  Unknown compiler version - please run the configure tests and report the results
  image_fallback.cc
  Unknown compiler version - please run the configure tests and report the results
  image_gif.cc
  Unknown compiler version - please run the configure tests and report the results
  image_jpg.cc
  Unknown compiler version - please run the configure tests and report the results
  image_png.cc
  Unknown compiler version - please run the configure tests and report the results
  Generating Code...
  process.vcxproj -> D:\Git\UFG\build\usd_from_gltf-master\process\Release\process.lib
  Building Custom Rule D:/Git/UFG/usd_from_gltf-master/convert/CMakeLists.txt
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/convert/CMakeFiles/generate.stamp is up-to-date.
  tokens.cc
  Unknown compiler version - please run the configure tests and report the results
  convert_util.cc
  Unknown compiler version - please run the configure tests and report the results
  converter.cc
  Unknown compiler version - please run the configure tests and report the results
D:\Git\USD\inst\USD\include\pxr/usd/usd/modelAPI.h(352): warning C4003: not enough arguments for function-like macro invocation 'BOOST_VMD_IS_TUPLE' [D:\Git\UFG\build\usd_from_gltf-master\convert\convert.vcxproj]
D:\Git\USD\inst\USD\include\pxr/usd/usd/modelAPI.h(352): warning C4003: not enough arguments for function-like macro invocation 'BOOST_VMD_DETAIL_IS_TUPLE' [D:\Git\UFG\build\usd_from_gltf-master\convert\convert.vcxproj]
D:\Git\USD\inst\USD\include\pxr/usd/usd/modelAPI.h(352): warning C4003: not enough arguments for function-like macro invocation 'BOOST_VMD_DETAIL_PARENS_SPLIT' [D:\Git\UFG\build\usd_from_gltf-master\convert\convert.vcxproj]
D:\Git\USD\inst\USD\include\pxr/usd/usd/modelAPI.h(352): warning C4003: not enough arguments for function-like macro invocation 'BOOST_PP_SEQ_DETAIL_IS_NOT_EMPTY' [D:\Git\UFG\build\usd_from_gltf-master\convert\convert.vcxproj]
D:\Git\USD\inst\USD\include\pxr/usd/usd/modelAPI.h(352): warning C4003: not enough arguments for function-like macro invocation 'BOOST_PP_SEQ_DETAIL_EMPTY_SIZE' [D:\Git\UFG\build\usd_from_gltf-master\convert\convert.vcxproj]
D:\Git\USD\inst\USD\include\pxr/usd/usd/modelAPI.h(352): warning C4003: not enough arguments for function-like macro invocation 'BOOST_PP_TUPLE_SIZE' [D:\Git\UFG\build\usd_from_gltf-master\convert\convert.vcxproj]
  materializer.cc
  Unknown compiler version - please run the configure tests and report the results
  package.cc
  Unknown compiler version - please run the configure tests and report the results
  texturator.cc
  Unknown compiler version - please run the configure tests and report the results
  Generating Code...
  convert.vcxproj -> D:\Git\UFG\build\usd_from_gltf-master\convert\Release\convert.lib
  Building Custom Rule D:/Git/UFG/usd_from_gltf-master/ufg_plugin/CMakeLists.txt
  Building Custom Rule D:/Git/UFG/usd_from_gltf-master/usd_from_gltf/CMakeLists.txt
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/ufg_plugin/CMakeFiles/generate.stamp is up-to-date.
  CMake does not need to re-run because D:/Git/UFG/build/usd_from_gltf-master/usd_from_gltf/CMakeFiles/generate.stamp is up-to-date.
  args.cc
  ufg_plugin.cc
  Unknown compiler version - please run the configure tests and report the results
  Unknown compiler version - please run the configure tests and report the results
  usd_from_gltf.cc
  Unknown compiler version - please run the configure tests and report the results
  Generating Code...
LINK : fatal error LNK1104: cannot open file 'boost_python3-vc141-mt-1_65_1.lib' [D:\Git\UFG\build\usd_from_gltf-master\usd_from_gltf\usd_from_gltf.vcxproj]
LINK : fatal error LNK1104: cannot open file 'boost_python3-vc141-mt-1_65_1.lib' [D:\Git\UFG\build\usd_from_gltf-master\ufg_plugin\ufg_plugin.vcxproj]

  USD_FROM_GLTF: ERROR: Command failed: cmake --build . --config Release --target install -- /m
  USD_FROM_GLTF: See log at: D:\Git\UFG\build\usd_from_gltf-master\log.txt

**Comment by C0DED00D** [comment]
Are you using the Visual Studio 2017 Developer Command Prompt for 64-bit (x64) when running the python script? 

Thanks!

**Comment by vanlicht** [comment]
@C0DED00D Yes, I use Visual Studio 2017 Developer Command Prompt for 64-bit (x64) for running the ufginstall.py

**Comment by lanxinger** [comment]
@C0DED00D same here.

**Comment by C0DED00D** [comment]
Ok, we'll add it to our backlog to investigate!

**Comment by lanxinger** [comment]
@C0DED00D thank you!

**Comment by hujasonx** [comment]
Hi all, sorry for the slow turnaround. I can't seem to recreate your issues on Windows 10 with Visual Studio 2017 and python 2.7.15

Did you use python 2 to install USD? Can you try changing "Python" to "Python2" on [line 21 of CmakeLists.txt](https://github.com/google/usd_from_gltf/blob/master/CMakeLists.txt#L21) and try again?

**Comment by vanlicht** [comment]
Hi,
Still can't built after changing the code in CmakeLists.txt as suggested. Here's the error log:
`cmake C:\UFG_usd_from_gltf\src\libpng-1.6.37 -DCMAKE_INSTALL_PREFIX=C:\UFG_usd_from_gltf -DCMAKE_PREFIX_PATH=C:\UFG_usd_from_gltf -DCMAKE_POSITION_INDEPENDENT_CODE=1 -DPNG_BUILD_ZLIB=yesplease -DZLIB_INCLUDE_DIR=C:\UFG_usd_from_gltf\src\zlib-1.2.11 -DCMAKE_PREFIX_PATH=C:\UFG_usd_from_gltf\src\zlib-1.2.11\Release -DZLIB_LIBRARY=C:\UFG_usd_from_gltf\src\zlib-1.2.11\Release\zlib.lib -DCMAKE_GENERATOR_PLATFORM=x64
-- Building for: Visual Studio 15 2017
-- The C compiler identification is MSVC 19.16.27034.0
-- The ASM compiler identification is MSVC
-- Found assembler: C:/Program Files (x86)/Microsoft Visual Studio/2017/BuildTools/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/BuildTools/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/BuildTools/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test HAVE_LD_VERSION_SCRIPT
-- Performing Test HAVE_LD_VERSION_SCRIPT - Failed
-- Performing Test HAVE_SOLARIS_LD_VERSION_SCRIPT
-- Performing Test HAVE_SOLARIS_LD_VERSION_SCRIPT - Failed
-- Configuring done
-- Generating done
-- Build files have been written to: C:/UFG_usd_from_gltf/build/libpng-1.6.37

cmake --build . --config Release --target install -- /m
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/UFG_usd_from_gltf/src/libpng-1.6.37/CMakeLists.txt
  Building Custom Rule C:/UFG_usd_from_gltf/src/libpng-1.6.37/CMakeLists.txt
  png-fix-itxt.c
  Building Custom Rule C:/UFG_usd_from_gltf/src/libpng-1.6.37/CMakeLists.txt
  Building Custom Rule C:/UFG_usd_from_gltf/src/libpng-1.6.37/CMakeLists.txt
LINK : fatal error LNK1181: cannot open input file 'C:\UFG_usd_from_gltf\src\zlib-1.2.11\Release\zlib.lib' [C:\UFG_usd_from_gltf\build\libpng-1.6.37\png-fix-itxt.vcxproj]
  png.c
  png.c
  pngerror.c
  pngerror.c
  pngget.c
  pngget.c
  pngmem.c
  pngmem.c
  pngpread.c
  pngpread.c
  pngread.c
  pngread.c
  pngrio.c
  pngrio.c
  pngrtran.c
  pngrtran.c
  pngrutil.c
  pngrutil.c
  pngset.c
  pngset.c
  pngtrans.c
  pngtrans.c
  pngwio.c
  pngwio.c
  pngwrite.c
  pngwrite.c
  pngwtran.c
  pngwtran.c
  pngwutil.c
  pngwutil.c
  Generating Code...
  Generating Code...
LINK : fatal error LNK1181: cannot open input file 'C:\UFG_usd_from_gltf\src\zlib-1.2.11\Release\zlib.lib' [C:\UFG_usd_from_gltf\build\libpng-1.6.37\png.vcxproj]
  png_static.vcxproj -> C:\UFG_usd_from_gltf\build\libpng-1.6.37\Release\libpng16_static.lib`

**Comment by hujasonx** [comment]
Hi, thanks for the response. Were there any interesting errors in the logs from earlier parts of the install process? Can you confirm that there is a 'C:\UFG_usd_from_gltf\src\zlib-1.2.11\Release\zlib.lib' file (There should be a .dll as well)? It should be generated in the preceding step in the install process.

Zlib should be being built automatically, but the steps you can go through to build it manually is to run
`cmake -DCMAKE_GENERATOR_PLATFORM=x64 .`
`cmake`
`msbuild /P:Configuration=Release zlib.sln`

Thanks for your patience and for helping out!

**Comment by fabriciochamon** [comment]
hi, any news on this one? I'm trying to build but also got:

LINK : fatal error LNK1104: cannot open file 'boost_python3-vc141-mt-1_65_1.lib' [D:\UFG\build\usd_from_gltf-master\usd_from_gltf\usd_from_gltf.vcxproj]
LINK : fatal error LNK1104: cannot open file 'boost_python3-vc141-mt-1_65_1.lib' [D:\UFG\build\usd_from_gltf-master\ufg_plugin\ufg_plugin.vcxproj]

running on vs2017 developer prompt, tried python2 and 3, changed cmakelists.txt and all the previous suggestions. Zlib is built correctly.

what to try next ? thank you!

**Comment by bitfoxgit** [comment]
The issues I ran into with building with Visual Studio 2017 and python 2.7:

1. zlib-1.2.11 required v142 build tools (so basically VS2019) I solved this by copying the command from the console window and building it manually in the 2019 console.
2 The "cannot open file 'boost_python3-vc141-mt-1_65_1.lib'": It looks like something is trying to find the newest python version and writes it to ".\usd_from_gltf\build\usd_from_gltf-master\usd_from_gltf\usd_from_gltf.vcxproj"
Removing all newer versions of python from path did not help. I renamed all python installations so only 2.7 remained and after that, the build went through.
I hope this helps anyone stumbling upon the same issues.

**Comment by mura94** [comment]
I know this is an old thread but I just ran into this issue. I'll just add to @bitsandfoxes 's comment above here for anyone else struggling to build this. Here are a few things that weren't mentioned in the installation guide that seem necessary at this stage:

- Check in log.txt (found in UFG\build\usd_from_gltf) to see which version of Python it found on your system. I had to delete/rename 4 versions of python above 2.7 that were in odd places like Visual Studio 2019 modules, Anaconda, or just regular installs of Python. 
  - (Kinda frustrating! Might want to include a disclaimer somewhere that you can't build UFG while you have a version of python >2.7 *anywhere* in your system )

- In its current state you can't build UFG with a USD build of a version greater than [19.05](https://github.com/PixarAnimationStudios/USD/releases/tag/v19.05). Spent quite a bit of time (and disk space) on this before I saw someone mention that elsewhere.

- Use the Visual Studio 2017 x64 Native Tools Command Prompt to build. Also saw that mentioned off-hand in another issue thread.

Hope that saves someone some time troubleshooting in the future. Thanks, @bitsandfoxes for saving me a ton of digging.

**Comment by ghost** [comment]
@bitsandfoxes @mura94 Thank you so much for the advice around getting this built. Such a mission, but glad I got there in the end. Just to add again to this conversation:

- I had to build the [19.05](https://github.com/PixarAnimationStudios/USD/releases/tag/v19.05) version of USD with x64_x86 Cross Platform Build Tool 2017, and I had to make sure I didn't have the newer version of USD in PATH because the older version wanted to use the newer version of Boost which caused ugly build errors.
- I also had to build zlib separately in a 2019 x64 Native Build Tools  environment and copy the zlib folder from the /src directory to the same directory in the 2017 x64 Native Build Tools environment.
- I also needed to make sure python3 was not available and only python 2.7
